lib/Parse/ParseDecl.cpp

```diff
@@ -5481,6 +5481,8 @@ ParserResult<EnumDecl> Parser::parseDeclEnum(ParseDeclOptions Flags,
                                         { }, nullptr, CurDeclContext);
   setLocalDiscriminator(ED);
   ED->getAttrs() = Attributes;
+  if (SF.Kind == SourceFileKind::Interface)
+    ED->setAddedImplicitInitializers();
```

Comment:
Hmm. I'm prefer not to set semantic information from the parser. How about performing the "is this decl in an interface source file?" check at the beginning of `TypeChecker::addImplicitConstructors()`?

Reply:
I thought about that, and it works, but this struck me as similar to what we do in the importer. I can still change it if you prefer, though.

Reply:
The importer is necessarily semantic, though: it’s bringing forth fully type-checked ASTs in the same way that module file deserialization does. 

The longer term idea for the parser is that it won’t build Decls at all.